tests/compile-fail/data_race/alloc_read_race.rs

```diff
@@ -29,7 +30,7 @@ pub fn main() {
             // Uses relaxed semantics to not generate
             // a release sequence.
             let pointer = &*ptr.0;
-            pointer.store(Box::into_raw(Box::new(MaybeUninit::uninit())), Ordering::Relaxed);
+            pointer.store(Box::into_raw(Box::new_uninit()), Ordering::Relaxed);
```

Comment:
@JCTyblaidd I had to change this test as (I think) a bug got fixed where the "write" of a `MaybeUninit::uninit()` into this box did not trigger the "write" hook.